osu.Game/Screens/Tournament/Drawings.cs

```diff
@@ -0,0 +1,367 @@
+                                }
+                            }
+                        }
+                    }
+                }
+            };
+
+            float offset = 0;
+            for (int i = 0; i < 6; i++)
+            {
+                visualiserContainer.Add(new VisualiserLine(2 * (float)Math.PI, offset, RNG.Next(10000, 12000))
+                {
+                    RelativeSizeAxes = Axes.Both,
+                });
+
+                offset += (float)Math.PI / 6f;
+            }
+
+            teamsContainer.OnSelected += t =>
+            {
+                groupsContainer.AddTeam(t.Team);
+
+                st.Text = t.Team.FullName;
+                st.FadeIn(200);
+
+                writeResults(groupsContainer.ToStringRepresentation());
+            };
+
+            teamsContainer.OnScrollStarted += () => st.FadeOut(200);
+
+            reset(true);
+        }
+
+        private void writeResults(string text)
+        {
+            lastWriteOp = writeOp;
+            writeOp = Task.Run(async () =>
+            {
+                if (lastWriteOp != null)
+                    await lastWriteOp;
+
+                // Write to drawings_results
+                try
+                {
+                    using (Stream stream = storage.GetStream(results_filename, FileAccess.Write, FileMode.Create))
+                    using (StreamWriter sw = new StreamWriter(stream))
+                    {
+                        sw.Write(text);
+                    }
+                }
+                catch
+                {
+                }
+            });
+        }
+
+        private void reloadTeams()
+        {
+            teamsContainer.ClearTeams();
+            allTeams.Clear();
+
+            try
+            {
+                using (Stream stream = storage.GetStream(teams_filename, FileAccess.Read, FileMode.Open))
+                using (StreamReader sr = new StreamReader(stream))
+                {
+                    while (sr.Peek() != -1)
+                    {
+                        string line = sr.ReadLine();
+
+                        string[] split = line.Split(':');
+                        string flagName = split[0].Trim();
+                        string name = split[1].Trim();
+
+                        string acronym = name.Substring(0, Math.Min(3, name.Length));
+                        if (split.Length >= 3)
+                            acronym = split[2].Trim();
+
+                        if (groupsContainer.ContainsTeam(name))
+                            continue;
+
+                        Team t = new Team()
+                        {
+                            FlagName = flagName,
+                            FullName = name,
+                            Acronym = acronym
+                        };
+
+                        allTeams.Add(t);
+                        teamsContainer.AddTeam(t);
+                    }
+                }
+            }
+            catch { }

```

Comment:
again, this should at very least error (and probably only be catching `IOException`?)